calendar-tool/tool/src/java/org/sakaiproject/calendar/entityproviders/CalendarEventEntityProvider.java

```diff
@@ -111,23 +121,62 @@ public boolean entityExists(final String id) {
 		// user being logged in and having access to the site is handled in the
 		// API
 		Calendar cal;
-		try {
-			cal = calendarService.getCalendar(String.format(
-					"/calendar/calendar/%s/main", siteId));
+		CalendarUtil calendarUtil;
+		//check parameter for merged in request
+		if(params.get("merged") != null && (params.get("merged")).equals("true")){
+			//request is to fetch all the events from merged calendars
+			try {
+				//get the current month value for the jquery calendar in lessons
+				long timestamp = NumberUtils.toLong((String)params.get("start"), 0);
+				if(timestamp == 0){
+					//set default to current date and month
+					calendarUtil = new CalendarUtil();
+				}
+				else{
+					java.util.Calendar calendar = java.util.Calendar.getInstance();
+					calendar.setTimeInMillis(timestamp * 1000L);
+					calendarUtil = new CalendarUtil(calendar);
+				}
+				//get all events from the merged calendars
+				CalendarEventVector calendarEventVector = calendarService.getEvents(calendarService.getCalendarReferences(siteId), getMonthTimeRange(calendarUtil));
 
-			for (Object o : cal.getEvents(null, null)) {
-				CalendarEvent event = (CalendarEvent) o;
+				Properties config = new Properties();
+				config.load(getClass().getResourceAsStream("/org/sakaiproject/calendar/tool/calendar.config"));
+				Map<String, String> eventImageMap = new CalendarUtil().getEventImageMap(config);
+				for (Object o : calendarEventVector) {
+					CalendarEvent event = (CalendarEvent) o;
 
-				r.add(new CalendarEventSummary(event));
+					CalendarEventSummary eventSummary = new CalendarEventSummary(event);
+					eventSummary.setEventImage(eventImageMap.get(event.getType()));
+					r.add(eventSummary);
+				}
+
+				return r;
+			}catch (IOException e) {
+				throw new EntityNotFoundException("Calendar config file missing: " + siteId,
+						siteId);
 			}
+		}
+		//fetch all the events from this site only
+		else{
+			try {
+				cal = calendarService.getCalendar(String.format(
+						"/calendar/calendar/%s/main", siteId));
 
-			return r;
-		} catch (IdUnusedException e) {
-			throw new EntityNotFoundException("Invalid siteId: " + siteId,
-					siteId);
-		} catch (PermissionException e) {
-			throw new EntityNotFoundException("No access to site: " + siteId,
-					siteId);
+				for (Object o : cal.getEvents(null, null)) {
```

Comment:
this was the original code present in the class and the trunk has similar code in the new file too.